gobblin-api/src/main/java/org/apache/gobblin/dataset/DatasetResolver.java

```diff
@@ -32,4 +35,20 @@
    * @return resolved dataset for the job
    */
   DatasetDescriptor resolve(DatasetDescriptor raw, State state);
+
+  @Override
```

Comment:
Why do you have the `@Override` annotation?

Reply:
It is added automatically by Intellij. Is it usually the case for interface method implementation to have `@Override` annotation?